crates/mun_codegen/src/db.rs

```diff
@@ -42,9 +41,15 @@ pub trait IrDatabase: hir::HirDatabase {
     #[salsa::invoke(crate::ir::ty::struct_ty_query)]
     fn struct_ty(&self, s: hir::Struct) -> StructType;
 
+    /// Given a `hir::FileId` generate code that is shared among the group of files.
+    /// TODO: Currently, a group always consists of a single file. Need to add support for multiple
+    /// files using something like `FileGroupId`.
+    #[salsa::invoke(crate::ir::file_group::ir_query)]
+    fn group_ir(&self, file: hir::FileId) -> Arc<FileGroupIR>;
+
     /// Given a `hir::FileId` generate code for the module.
-    #[salsa::invoke(crate::ir::module::ir_query)]
-    fn module_ir(&self, file: hir::FileId) -> Arc<ModuleIR>;
+    #[salsa::invoke(crate::ir::file::ir_query)]
+    fn module_ir(&self, file: hir::FileId) -> Arc<FileIR>;
```

Comment:
Maybe then rename this method to `file_ir` as well?

Reply:
Oops. Will do